### ISSUE TITLE
Problems running on cs3e Hebbe

### ISSUE THREAD

**Issue by JohnHellgren** [SciLifeLab/NGI-RNAseq#174]
Hello,
I'm trying to run the pipeline on Hebbe (C3SE), but I have some problems with the configuration, which makes the pipeline fail at the process genebody_coverage and markDuplicates due to memory problems.

I can tell you straight away that I have limited linux and programming skills, but I'll try to explain the problem anyway. 

The pipeline gives me this error:
```
[35/e80538] NOTE: Error submitting process 'genebody_coverage (NC1_S1_L002_R1_001AlignedByCoord.out)' for execution -- Error is ignored

ERROR ~ Error executing process > 'markDuplicates (NC1_S1_L002_R1_001AlignedByCoord.out)'

Caused by:
  Failed to submit process to grid scheduler for execution

Command executed:

  sbatch .command.run

Command exit status:
  1

Command output:
  sbatch: error: rejecting job, too much memory requested for 2 cores. You must use -C MEMXXX or increase number of requested cores instead of using --mem: pn_min_memory = 16384
  sbatch: error: Batch job submission failed: Invalid generic resource (gres) specification"
```

This is my configuration file (from this [guide](https://wabi-wiki.scilifelab.se/display/KB/Install+RNA-seq+pipeline+on+hebbe))

[config_TEMPLATE.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1546974/config_TEMPLATE.txt)

I've had the similar problem with starting trim galore, but that was fixed by adding cpus=8 for trim galore in the config file. I tried to do the same for dupradar (not sure that it's connected to mark_Duplicates), but that gave the same error. 
I also tried to use [this](https://github.com/ewels/NGI-RNAseq/blob/master/conf/hebbe.config) config file instead, but that could not start trim galore due to memory problems.

Here's my nextflow script:
[rna_seq_script.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1547059/rna_seq_script.txt)

I submitted the job using this command line:
```
sbatch -C "MEM64" rna_seq_script.sh
```

log files:
[NGI-RNAseq_trace.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1547063/NGI-RNAseq_trace.txt)
[nextflow.log](https://github.com/SciLifeLab/NGI-RNAseq/files/1547081/nextflow.log)
[mystdout.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1547083/mystdout.txt)

Any suggestions?

Best regards,
John Hellgren

**Comment by Hammarn** [comment]
It looks like Nextflow is using the wrong format for requesting memory. In line with what the scheduler is telling you I think you should only request cores instead of memory.
Remove all the lines that are requesting memory in the [base.config](https://github.com/SciLifeLab/NGI-RNAseq/blob/master/conf/base.config).
I think that should solve your problem. Let me know how it works.

**Comment by JohnHellgren** [comment]
Hello, 
That config file did not work:
"ERROR ~ Error executing process > 'trim_galore (NC1_S1_L001_R)'

Caused by:
  No signature of method: check_max() is applicable for argument types: (java.lang.Integer, java.lang.String) values: [2, cpus]
Possible solutions: check_log(java.lang.Object) "

However, I added this to my previous config file:

  $genebody_coverage {
    cpus = 8
  }
   $markDuplicates {
	cpus = 8

But now I have a new error instead of the memory error:
ERROR ~ Error executing process > 'markDuplicates (NC1_S1_L002_R1_001AlignedByCoord.out)'

Caused by:
  Process `markDuplicates (NC1_S1_L002_R1_001AlignedByCoord.out)` terminated with an error exit status (1)

Command executed:

  java -Xmx16g -jar $PICARD_HOME/picard.jar MarkDuplicates \
      INPUT=NC1_S1_L002_R1_001Aligned.sortedByCoord.out.bam \
      OUTPUT=NC1_S1_L002_R1_001Aligned.sortedByCoord.out.markDups.bam \
      METRICS_FILE=NC1_S1_L002_R1_001Aligned.sortedByCoord.out.markDups_metrics.txt \
      REMOVE_DUPLICATES=false \
      ASSUME_SORTED=true \
      PROGRAM_RECORD_ID='null' \
      VALIDATION_STRINGENCY=LENIENT

  # Print version number to standard out
  echo "File name: NC1_S1_L002_R1_001Aligned.sortedByCoord.out.bam Picard version "$(java -Xmx2g -jar $PICARD_HOME/picard.jar  MarkDuplicates --version 2>&1)
  samtools index NC1_S1_L002_R1_001Aligned.sortedByCoord.out.bam

Command exit status:
  1

Command output:
  (empty)

Command error:
  Error: Unable to access jarfile /c3se/NOBACKUP/users/johnhe/Hebbe/miniconda3/envs/rna_seq_py2.7/share/picard-2.9.0-0//picard.jar

**Comment by Hammarn** [comment]
Yeah, we have a bunch of logic that check the memory requirements, those would need to be ripped out as well for that to work.

Your new error does _look_ simple enough. Can you access the picard jar at that path?

**Comment by JohnHellgren** [comment]
hmm... This double slash looks wrong...
picard-2.9.0-0//picard.jar

**Comment by Hammarn** [comment]
Unix systems doesn't care about double slashes in paths so I think that's fine - as long as your user has access to it it should be fine.

**Comment by JohnHellgren** [comment]
Alright
The path was wrong, should be 
/c3se/NOBACKUP/users/johnhe/miniconda3/envs/rna_seq_py2.7/share/picard-2.9.0-0/picard.jar

No extra Hebbe

**Comment by ewels** [comment]
Hi @JohnHellgren,

Sorry for the problems you're having - Jakub's NBIS guide is quite old now and a lot in the pipeline has changed since then. Specifically, I've refactored much of how the config files work and updated the hebbe profile. However, as we don't have access to Hebbe (and Jakub's project has finished) I had no way to test these changes.

We recently updated the pipeline to use Singularity on UPPMAX, which handles all of the software requirements without you having to do anything. It looks like Hebbe also has Singularity installed (according to [this page](http://www.c3se.chalmers.se/index.php/Singularity)).

If this is the case, I think that I should be able to update the hebbe config to work properly for you so that you don't need to do anything special to make the pipeline run.

I'll whip together a quick pull request for you now and we can see if that is any better.

Phil

**Comment by ewels** [comment]
Hi @JohnHellgren,

I've just made a pull-request with changes which will hopefully solve your configuration problems on hebbe (#175). This will now use singularity, so everything to do with software installation from Jakub's notes is not needed. I've also removed all memory requests, so _hopefully_ you won't get any errors from that (though we shall see!).

Once @Hammarn merges the pull requests, please pull the latest version of the pipeline:

```
nextflow pull SciLifeLab/NGI-RNAseq
```

Then, make a detached login so that you can close your terminal window without killing running jobs (eg. I use `tmux` to do this - tutorial [here](https://danielmiessler.com/study/tmux/). For example:

```bash
tmux new -s rna_analysis
```

Now, launch the pipeline. Note that Nextflow launches the `sbatch` jobs for you, so you don't need to run nextflow in an sbatch job (though you can if you want, as you found out above).

Note that all paths are relative, and it's generally good to run nextflow in the directory where you have your project. Then it will create the work directory there with the intermediate files that you can delete afterwards or use when resuming.

I've put together a launch command for you based on what you've written above:

```bash
cd /c3se/NOBACKUP/users/johnhe/rnaseq/

nextflow run SciLifeLab/NGI-RNAseq \
    -profile hebbe \
    -with-dag flowchart.pdf \
    --project SNIC2017-1-89 \
    --reads "raw/*{1,2}_001.fastq.gz" \
    --outdir "results3" \
    --bed12 "../referencegenome/genes.bed12" \
    --fasta "../referencegenome/genome.fa" \
    --gtf "../referencegenome/genes.gtf" \
    --star "../referencegenome/star"
```

Now - sit back and wait for any errors! Hopefully there won't be and it will run first time (fingers crossed!) and once you've waited a minute or two and are satisfied you can close your window. If using `tmux` you do this by doing `ctrl+b` followed by `d` for detach.

If you want to check the progress of your run later, you can re-attach to the session using:

```
tmux a -t rna_analysis
```

(or just `tmux a` if you only have one session running).

If this works, we can move much of that nextflow command into a config file for you to make launching the pipeline. For example, if you create a file in your home directory at `~/.nextflow/config` that looks like this:

```groovy
params {
  project = "SNIC2017-1-89"
  genomes {
    'R64-1-1' {
      bed12   = '/c3se/NOBACKUP/users/johnhe/referencegenome/genes.bed12'
      fasta   = '/c3se/NOBACKUP/users/johnhe/referencegenome/genome.fa'
      gtf     = '/c3se/NOBACKUP/users/johnhe/referencegenome/genes.gtf'
      star    = '/c3se/NOBACKUP/users/johnhe/referencegenome/star'
    }
  }
```

Then your nextflow command can be the somewhat simpler:

```bash
cd /c3se/NOBACKUP/users/johnhe/rnaseq/

nextflow run SciLifeLab/NGI-RNAseq \
    -profile hebbe \
    -with-dag flowchart.pdf \
    --genome 'R64-1-1' \
    --reads "raw/*{1,2}_001.fastq.gz" \
    --outdir "results3"
```

Note that specifying your email with `--email` or in the config file makes life easier as you will be notified when the run is finished.

Ok, I hope that all helps! Let us know how you get on.

Phil

**Comment by JohnHellgren** [comment]
I'm not sure if I need to make any additional changed to use the singularity

```
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/c3se/NOBACKUP/users/johnhe/miniconda3/envs/rna_seq_py2.7/share/picard-2.9.0-0/tmp
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/c3se/NOBACKUP/users/johnhe/miniconda3/envs/rna_seq_py2.7/share/picard-2.9.0-0/tmp
N E X T F L O W  ~  version 0.26.3
Launching `SciLifeLab/NGI-RNAseq` [curious_monod] - revision: fae0bd2b71 [master]
=========================================
 NGI-RNAseq : RNA-Seq Best Practice v1.4dev
=========================================
Run Name       : curious_monod
Reads          : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/*{1,2}_001.fastq.gz
Data Type      : Paired-End
Genome         : R64-1-1
Strandedness   : None
Trim R1        : 0
Trim R2        : 0
Trim 3' R1     : 0
Trim 3' R2     : 0
Aligner        : STAR
STAR Index     : /c3se/NOBACKUP/users/johnhe/referencegenome/star
GTF Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.gtf
BED Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.bed12
Save Reference : Yes
Save Trimmed   : No
Save Intermeds : No
Max Memory     : 64 GB
Max CPUs       : 20
Max Time       : 10d
Output dir     : /c3se/NOBACKUP/users/johnhe/rnaseq/results2
Working dir    : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work
Container      : docker://scilifelab/ngi-rnaseq:latest
Pipeline Release: master
Current home   : /c3se/users/johnhe/Hebbe
Current user   : johnhe
Current path   : /c3se/NOBACKUP/users/johnhe/rnaseq/raw
R libraries    : false
Script dir     : /c3se/users/johnhe/Hebbe/.nextflow/assets/SciLifeLab/NGI-RNAseq
Config Profile : hebbe
UPPMAX Project : SNIC2017-1-89
=========================================
[warm up] executor > slurm
WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/singularity
Pulling Singularity image docker://scilifelab/ngi-rnaseq:latest [cache /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/singularity/scilifelab-ngi-rnaseq-latest.img]
[warm up] executor > local
[02/af3334] Submitted process > trim_galore (NC1_S1_L002_R)
[ec/aeb137] Submitted process > fastqc (NC1_S1_L002_R)
ERROR ~ Error executing process > 'trim_galore (NC1_S1_L002_R)'

Caused by:
  Process `trim_galore (NC1_S1_L002_R)` terminated with an error exit status (255)

Command executed:

  trim_galore --paired --fastqc --gzip     NC1_S1_L002_R1_001.fastq.gz NC1_S1_L002_R2_001.fastq.gz

Command exit status:
  255

Command output:
  (empty)

Command error:
  [91mERROR  : Image path doesn't exists
  [0m[31mABORT  : Retval = 255
  [0m

Work dir:
  /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/02/af3334de0f0f3ce5fcd480ef2902cb

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`

 -- Check '.nextflow.log' file for details
Execution cancelled -- Finishing pending tasks before exit
[ec/aeb137] NOTE: Process `fastqc (NC1_S1_L002_R)` terminated with an error exit status (255) -- Error is ignored
[NGI-RNAseq] Pipeline Complete
```

**Comment by ewels** [comment]
Ooh, that's a new one. Singularity is failing to pull the image from the internet for some reason. Does hebbe have an internet connection?

Are you able to manually create the singularity image file? This is the command:

```
singularity pull --name scilifelab-ngi-rnaseq-latest.img docker://scilifelab/ngi-rnaseq:latest
```

If that works, you should have a new singularity image file in your working directory. You can then instruct nextflow to use that for the pipeline by using the additional command line flag:

```
-with-singularity scilifelab-ngi-rnaseq-latest.img
```

Phil

**Comment by JohnHellgren** [comment]
That worked, but I'm getting stuck at another place instead...

```
N E X T F L O W  ~  version 0.26.3
Launching `SciLifeLab/NGI-RNAseq` [sleepy_ride] - revision: fae0bd2b71 [master]
=========================================
 NGI-RNAseq : RNA-Seq Best Practice v1.4dev
=========================================
Run Name       : sleepy_ride
Reads          : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/*{1,2}_001.fastq.gz
Data Type      : Paired-End
Genome         : R64-1-1
Strandedness   : None
Trim R1        : 0
Trim R2        : 0
Trim 3' R1     : 0
Trim 3' R2     : 0
Aligner        : STAR
STAR Index     : /c3se/NOBACKUP/users/johnhe/referencegenome/star
GTF Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.gtf
BED Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.bed12
Save Reference : Yes
Save Trimmed   : No
Save Intermeds : No
Max Memory     : 64 GB
Max CPUs       : 20
Max Time       : 10d
Output dir     : /c3se/NOBACKUP/users/johnhe/rnaseq/results2
Working dir    : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work
Container      : scilifelab-ngi-rnaseq-latest.img
Pipeline Release: master
Current home   : /c3se/users/johnhe/Hebbe
Current user   : johnhe
Current path   : /c3se/NOBACKUP/users/johnhe/rnaseq/raw
R libraries    : false
Script dir     : /c3se/users/johnhe/Hebbe/.nextflow/assets/SciLifeLab/NGI-RNAseq
Config Profile : hebbe
UPPMAX Project : SNIC2017-1-89
=========================================
[warm up] executor > slurm
[2c/083bc8] Submitted process > trim_galore (NC1_S1_L002_R)
[warm up] executor > local
[c9/1f8fc1] Submitted process > fastqc (NC1_S1_L002_R)
ERROR ~ Error executing process > 'trim_galore (NC1_S1_L002_R)'

Caused by:
  Process `trim_galore (NC1_S1_L002_R)` terminated with an error exit status (127)

Command executed:

  trim_galore --paired --fastqc --gzip     NC1_S1_L002_R1_001.fastq.gz NC1_S1_L002_R2_001.fastq.gz

Command exit status:
  127

Command output:
  (empty)

Command error:
  ESC[33mWARNING: Non existent bind point (directory) in container: '/c3se'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/apps'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/usr/share/lmod/lmod'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/var/hasplm'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/local'
  ESC[0mESC[33mWARNING: Could not chdir to home: /c3se/users/johnhe/Hebbe
  ESC[0m/bin/bash: line 0: cd: /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/2c/083bc84b5ad6170088402d2641c0b3: No such file or directory
  /bin/bash: .command.run.1: No such file or directory

Work dir:
  /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/2c/083bc84b5ad6170088402d2641c0b3

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
```

**Comment by ewels** [comment]
Ahaa, yes of course. Sorry - I read about this in the c3se singularity docs and immediately forgot about it.

This requires me to update the docker image - should be a quick code change, but you'll need to wait for the public docker image to rebuild which takes a few hours sorry. I'll do the pull request now, then once the rebuild is complete you can delete and pull the singularity image again and hopefully it'll work.

Docker build status here: https://hub.docker.com/r/scilifelab/ngi-rnaseq/builds/

Phil

**Comment by JohnHellgren** [comment]
Looks like I get a similar error anyway.
Does singularity need additional configurations on hebbe?
```
N E X T F L O W  ~  version 0.26.3
Launching `SciLifeLab/NGI-RNAseq` [tender_yalow] - revision: fae0bd2b71 [master]
=========================================
 NGI-RNAseq : RNA-Seq Best Practice v1.4dev
=========================================
Run Name       : tender_yalow
Reads          : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/*{1,2}_001.fastq.gz
Data Type      : Paired-End
Genome         : R64-1-1
Strandedness   : None
Trim R1        : 0
Trim R2        : 0
Trim 3' R1     : 0
Trim 3' R2     : 0
Aligner        : STAR
STAR Index     : /c3se/NOBACKUP/users/johnhe/referencegenome/star
GTF Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.gtf
BED Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.bed12
Save Reference : Yes
Save Trimmed   : No
Save Intermeds : No
Max Memory     : 64 GB
Max CPUs       : 20
Max Time       : 10d
Output dir     : /c3se/NOBACKUP/users/johnhe/rnaseq/results2
Working dir    : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work
Container      : scilifelab-ngi-rnaseq-latest.img
Pipeline Release: master
Current home   : /c3se/users/johnhe/Hebbe
Current user   : johnhe
Current path   : /c3se/NOBACKUP/users/johnhe/rnaseq/raw
R libraries    : false
Script dir     : /c3se/users/johnhe/Hebbe/.nextflow/assets/SciLifeLab/NGI-RNAseq
Config Profile : hebbe
UPPMAX Project : SNIC2017-1-89
=========================================
[warm up] executor > slurm
[warm up] executor > local
[36/0150e3] Submitted process > trim_galore (NC1_S1_L002_R)
[0a/8c60ea] Submitted process > fastqc (NC1_S1_L002_R)
ERROR ~ Error executing process > 'trim_galore (NC1_S1_L002_R)'

Caused by:
  Process `trim_galore (NC1_S1_L002_R)` terminated with an error exit status (127)

Command executed:

  trim_galore --paired --fastqc --gzip     NC1_S1_L002_R1_001.fastq.gz NC1_S1_L002_R2_001.fastq.gz

Command exit status:
  127

Command output:
  (empty)

Command error:
  ESC[33mWARNING: Non existent bind point (directory) in container: '/c3se'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/apps'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/usr/share/lmod/lmod'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/var/hasplm'
  ESC[0mESC[33mWARNING: Non existent bind point (directory) in container: '/local'
  ESC[0mESC[33mWARNING: Could not chdir to home: /c3se/users/johnhe/Hebbe
  ESC[0m/bin/bash: line 0: cd: /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/36/0150e382407b5b9b40845ef6bde127: No such file or directory
  /bin/bash: .command.run.1: No such file or directory

Work dir:
  /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/36/0150e382407b5b9b40845ef6bde127

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
Execution cancelled -- Finishing pending tasks before exit
[0a/8c60ea] NOTE: Process `fastqc (NC1_S1_L002_R)` terminated with an error exit status (127) -- Error is ignored
[NGI-RNAseq] Pipeline Complete
WARN: To render the execution DAG in the required format it is required to install Graphviz -- See http://www.graphviz.org for more info.
```

**Comment by ewels** [comment]
Ah, @Hammarn didn't merge my pull-request yet (#177) so the docker image hasn't updated yet. I'll self-merge it now.

**Comment by JohnHellgren** [comment]
Ok, so I should wait until the new docker image is ready?
It says that it's queued [here](https://hub.docker.com/r/scilifelab/ngi-rnaseq/builds/)

**Comment by ewels** [comment]
Yes exactly - the build [failed](https://hub.docker.com/r/scilifelab/ngi-rnaseq/builds/bsfmrmpgqtnqclwmhycmzhc/) because I had a problem in the changes that I made though sorry. Hopefully just fixed that and [restarted the build now](https://hub.docker.com/r/scilifelab/ngi-rnaseq/builds/bd5tqmwqhml7wgxow3aovg3/).

**Comment by ewels** [comment]
Hi @JohnHellgren - the latest docker build has completed successfully. Hopefully it should work for you now 🤞

**Comment by JohnHellgren** [comment]
Yea, it looks like it's running now!
Fastqc and Trim galore is done!

**Comment by JohnHellgren** [comment]
No multiQC report though
I pulled the .img file at 17:00, did you do any changes after that?

```
N E X T F L O W  ~  version 0.26.3
Launching `SciLifeLab/NGI-RNAseq` [cranky_williams] - revision: f157ecf6e1 [master]
=========================================
 NGI-RNAseq : RNA-Seq Best Practice v1.4dev
=========================================
Run Name       : cranky_williams
Reads          : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/*{1,2}_001.fastq.gz
Data Type      : Paired-End
Genome         : R64-1-1
Strandedness   : None
Trim R1        : 0
Trim R2        : 0
Trim 3' R1     : 0
Trim 3' R2     : 0
Aligner        : STAR
STAR Index     : /c3se/NOBACKUP/users/johnhe/referencegenome/star
GTF Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.gtf
BED Annotation : /c3se/NOBACKUP/users/johnhe/referencegenome/genes.bed12
Save Reference : Yes
Save Trimmed   : No
Save Intermeds : No
Max Memory     : 64 GB
Max CPUs       : 20
Max Time       : 10d
Output dir     : /c3se/NOBACKUP/users/johnhe/rnaseq/results2
Working dir    : /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work
Container      : scilifelab-ngi-rnaseq-latest.img
Pipeline Release: master
Current home   : /c3se/users/johnhe/Hebbe
Current user   : johnhe
Current path   : /c3se/NOBACKUP/users/johnhe/rnaseq/raw
R libraries    : false
Script dir     : /c3se/users/johnhe/Hebbe/.nextflow/assets/SciLifeLab/NGI-RNAseq
Config Profile : hebbe
UPPMAX Project : SNIC2017-1-89
=========================================
[warm up] executor > slurm
[warm up] executor > local
[80/e49bf7] Submitted process > fastqc (NC1_S1_L002_R)
[e3/a6a937] Submitted process > trim_galore (NC1_S1_L002_R)
[23/607b86] Submitted process > star (NC1_S1_L002_R1_001)
          Passed alignment > star (NC1_S1_L002_R1_001)   >> 88.74% <<
[Picard MarkDuplicates] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.
[f5/7e57c0] Submitted process > rseqc (NC1_S1_L002_R1_001AlignedByCoord.out)
[f8/6c96aa] Submitted process > preseq (NC1_S1_L002_R1_001AlignedByCoord.out)
[13/b46dca] Submitted process > stringtieFPKM (NC1_S1_L002_R1_001AlignedByCoord.out)
[a7/dc2722] Submitted process > genebody_coverage (NC1_S1_L002_R1_001AlignedByCoord.out)
[18/45039e] Submitted process > markDuplicates (NC1_S1_L002_R1_001AlignedByCoord.out)
[07/85b7bc] Submitted process > featureCounts (NC1_S1_L002_R1_001AlignedByCoord.out)
[ad/e7aeca] Submitted process > merge_featureCounts (NC1_S1_L002_R1_001AlignedByCoord.out_gene.featureCounts)
[ce/32c24d] Submitted process > dupradar (NC1_S1_L002_R1_001Aligned.sortedByCoord.out.markDups)
[85/ee203a] Submitted process > get_software_versions
[NGI-RNAseq] Pipeline Complete
WARN: To render the execution DAG in the required format it is required to install Graphviz -- See http://www.graphviz.org for more info.
```

**Comment by ewels** [comment]
Ok great! Now we're getting somewhere..

MultiQC not running isn't a problem with the docker container, as it would try to run the process and then exit with an error. Here the process isn't even being kicked off by Nextflow.

I guess that this is because you're only running with one sample. When there are less than three samples, the correlation scripts are skipped and perhaps this also prevents MultiQC from running. Can you try running the pipeline with three or more samples?

Phil

**Comment by JohnHellgren** [comment]
It works now!
I ran the pipeline with all my samples and all looks good at the moment

[mystdout.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1554593/mystdout.txt)
[nextflowlog.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1554594/nextflowlog.txt)
[NGI-RNAseq_trace.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1554597/NGI-RNAseq_trace.txt)

Thanks a lot for your help!

**Comment by JohnHellgren** [comment]
The multiQC report did not include the following files (I can find them in the result folder though):

* Inner Distance
* Junction Saturation
* Read Duplication

**Comment by ewels** [comment]
Interesting! What does the MultiQC working directory look like? `work/34/a97f00*/rseqc `

Note MultiQC doesn't use the PDF files in the results output, it needs the `.txt` and `.r` files.

At the moment you won't get the sample correlation plots, dupRadar plot or Biotype counts plot in MultiQC as they're generated by our custom MultiQC plugin: [MultiQC_NGI](https://github.com/ewels/MultiQC_NGI). However, the currently open pull request #143 by @Hammarn which is waiting for me to review and merge updates the output from these steps so that MultiQC can pick them up without using the plugin. After I merge this and you pull updates again they should appear for you.

Phil

**Comment by JohnHellgren** [comment]
I downloaded the new [build](https://hub.docker.com/r/scilifelab/ngi-rnaseq/builds/bepxdwayantucub6ygsj2wd/) and updated the pipeline to revision: a891a45a74 [master], but now I ran into some memory problems with markDuplicates, so the pipeline could not finish

```
Exception in thread "main" htsjdk.samtools.util.RuntimeIOException: java.io.IOException: No space left on device
```

Sound very unlikely that I'm out of storage space (I'm using 161 GB out of the 200 GB soft quota)
Does it mean RAM?

You have the logs below:
[mystdout.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1559881/mystdout.txt)
[nextflowlog.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1559882/nextflowlog.txt)
[NGI-RNAseq_trace.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1559883/NGI-RNAseq_trace.txt)

**Comment by ewels** [comment]
Hi @JohnHellgren,

I guess this is due to the temporary directory that Picard is using. I think it should default to using the system default tmp directory (as found by Java). We could perhaps tell Picard to use the work directory for storing temporary files instead, which may help...

Phil

**Comment by JohnHellgren** [comment]
Hello again @ewels, 

I tried to set my own tmpdir for java:

```
export _JAVA_OPTIONS=-Djava.io.tmpdir=/c3se/NOBACKUP/users/johnhe/rnaseq/raw/javatmp
```
but I still get the same error (no space left on device) with latest build and revision (yesterday).

Any ideas?
You have the logs below:
[mystdout.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1618343/mystdout.txt)
[nextflowlog.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1618347/nextflowlog.txt)
[NGI-RNAseq_trace.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1618348/NGI-RNAseq_trace.txt)
[rna_seq_script.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1618351/rna_seq_script.txt)
[command.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1618352/command.txt)

The script used 1 node with 20 cores

Best regards,
John

**Comment by ewels** [comment]
Hmm, sorry John - this is pretty hard to help with as we can't replicate the error and don't have access to the cluster. It sounds like a problem with picard, rather than anything nextflow or pipeline related. Do you get the same error if you run `.command.sh` yourself manually?

**Comment by JohnHellgren** [comment]
Hello again,
I could make the pipeline finish by using the ```-resume``` command today on yesterdays run.
Maybe the tmpdir was cleared, which made it possible for the pipeline to finish the remaining processes?

The results looks good, however, still missing this in the multiQC:

* Inner Distance
* Junction Saturation
* Read Duplication
* Gene body coverage

I did not know exactly what to look for in ```/work/02/ca8b1c75fb927520fcc1501e0c27b9/rseqc```
There are a lot of logs: 
genebody_1
...
genebody_144

rseqc_log.1
...
rseqc_log.648

Here's the multiQC log (if it helps):

[multiqc.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1621968/multiqc.txt)

The sample correlation plots, dupRadar plot and Biotype counts plot is there now :thumbsup:

Best regards,
John

**Comment by ewels** [comment]
Aha, great spot - thanks! Yes I get the same thing with the test data. This is due to a fairly recent change by @Hammarn where we told Nextflow to rename files to avoid logs being accidentally overwritten. This has the unfortunate effect of disrupting the filename patterns that MultiQC uses to find the files though.. oops!

We'll sort this out asap, thanks for telling us.

**Comment by ewels** [comment]
Fixed in #191 

Thanks again for this thread - let us know if you run into anything else!

**Comment by JohnHellgren** [comment]
Hello again,
I'm about to run the pipeline again with some new data, but now I'm experiencing problems again (I updated to version 1.4 before running with the new samples)

``` 
[rna_seq_script.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1970266/rna_seq_script.txt)
Error executing process > 'trim_galore (NC1_S25_L001_R)'

Caused by:
  Process `trim_galore (NC1_S25_L001_R)` terminated with an error exit status (127)

Command executed:

  trim_galore --paired --fastqc --gzip     NC1_S25_L001_R1_001.fastq.gz NC1_S25_L001_R2_001.fastq.gz

Command exit status:
  127

Command output:
  (empty)

Command error:
  [33mWARNING: Non existent bind point (directory) in container: '/c3se'
  [0m[33mWARNING: Non existent bind point (directory) in container: '/apps'
  [0m[33mWARNING: Non existent bind point (directory) in container: '/usr/share/lmod/lmod'
  [0m[33mWARNING: Non existent bind point (directory) in container: '/var/hasplm'
  [0m[33mWARNING: Non existent bind point (directory) in container: '/local'
  [0m[33mWARNING: Could not chdir to home: /c3se/users/johnhe/Hebbe
  [0m/bin/bash: line 0: cd: /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/f9/4bd843218ecedb7cf8f2fc357253d4: No such file or directory
  /bin/bash: .command.stub: No such file or directory

Work dir:
  /c3se/NOBACKUP/users/johnhe/rnaseq/raw/work/f9/4bd843218ecedb7cf8f2fc357253d4

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh
````

Is it something with the Singularity setup?
It worked with the previous version of the pipeline

[rna_seq_script.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1970267/rna_seq_script.txt)
[mystdout.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1970268/mystdout.txt)
[nextflow.txt](https://github.com/SciLifeLab/NGI-RNAseq/files/1970269/nextflow.txt)

**Comment by ewels** [comment]
Hi @JohnHellgren,

This is because we removed the Hebbe-specific folder paths in the singularity containers in this release (plus those for several other clusters). The intention was that everyone should have overlays enabled in their singularity installation. This means that we don't need to customise the code towards every different computational system. This is an issue for the Hebbe sysadmins really, as it's related to the core singularity installation. See http://singularity.lbl.gov/docs-mount

Hopefully they can enable overlays, though older versions of linux are unfortunately not able to support this. Let me know if they say that it can't be done and I can try to help (I need to write some docs on this anyway).

Apologies that there wasn't more warning - I didn't think that this would affect many people, but I've now heard from a few so perhaps I underestimated how many people are running on old versions of linux 😕 

Phil

**Comment by JohnHellgren** [comment]
Hello,

They can't enable overlays:

"Hi John

Unfortunately, this relies on a relatively new kernel feature, overlayfs, which is not supported in the CentOS6 kernel (and upgrading is far from a simple operation).
This is also not a simple feature; enabling overlay caused a critical security hole in singularity 2.4.* just a few days ago (this feature was off by default for a reason i suspect) But, either way, we can't enable it.
For Hebbe (and Glenn), you have no option but to patch in relevant directories manually:
https://www.c3se.chalmers.se/documentation/software/#c3se-specific-additions

Best regards, Mikael"

**Comment by JohnHellgren** [comment]
How do I add those directories manually to the .img file?

**Comment by MaxUlysse** [comment]
For more and easier reproducibility on your side, I would make a specific [singularity recipe](https://singularity.lbl.gov/docs-recipes) and add the wanted directory when building the new one.

**Comment by ewels** [comment]
I'm closing this issue now as hopefully it's resolved, plus we're now moving over to [nf-core/rnaseq](https://github.com/nf-core/rnaseq) and will be archiving this repository.

If you have any remaining problems, please create a new issue on the above repo.
